packages/validator/src/services/syncingStatusTracker.ts

```diff
@@ -0,0 +1,60 @@
+import {ApiClient, routes} from "@lodestar/api";
+import {Logger} from "@lodestar/utils";
+import {Slot} from "@lodestar/types";
+import {IClock} from "../util/clock.js";
+
+export type SyncingStatus = routes.node.SyncingStatus;
+
+type RunOnSyncedFn = (slot: Slot, signal: AbortSignal) => Promise<void>;
+
+/**
+ * Track the syncing status of connected beacon node
+ */
+export class SyncingStatusTracker {
+  private prevSyncingStatus: SyncingStatus | null = null;
+
+  private readonly fns: RunOnSyncedFn[] = [];
+
+  constructor(
+    private readonly logger: Logger,
+    private readonly api: ApiClient,
+    private readonly clock: IClock
+  ) {
+    this.clock.runEverySlot(this.checkSyncingStatus);
+  }
+
+  /** Run function when node status changes from syncing to synced */
+  runOnSynced(fn: RunOnSyncedFn): void {
```

Comment:
maybe name this runOnResynced

Reply:
yeah, was wondering how this can be more specific to indicate that it's only if status changed to synced, I like your suggestion